compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -503,7 +503,7 @@ private STNode parseCompUnit() {
         boolean processImports = true;
         while (token.kind != SyntaxKind.EOF_TOKEN) {
             STNode decl = parseTopLevelNode(token.kind);
-            if (decl.kind == SyntaxKind.IMPORT_DECLARATION) {
+            if (decl != null && decl.kind == SyntaxKind.IMPORT_DECLARATION) {
```

Comment:
Shall we do this null check at the beginning and break the loop?

Reply:
sure